### ISSUE TITLE
Investigate 3to2

### ISSUE THREAD

**Issue by jayvdb** [coala/coala#3000]
https://pypi.python.org/pypi/3to2 downgrades code to Python 2. Maybe it is good enough to allow coala to run natively on Python 2, so that Python bears can run natively in the Python 2 runtime.

**Comment by sils** [comment]
I'm pretty sure that's not possible because we use python 3 only *syntax* like annotations.

**Comment by sils** [comment]
I'd rather think about including a 2to3 bear to let people upgrade their code more easier.

**Comment by sils** [comment]
(for this release?)

**Comment by jayvdb** [comment]
> I'm pretty sure that's not possible because we use python 3 only syntax like annotations.

I expect 3to2 would discard them, but we could store them somewhere else for use by Python 2 code.
This is just an 'investigate' task for me, as I am rather curious if it is possible.

fwiw, 2to3 sounds like a great idea for a bear.

**Comment by jayvdb** [comment]
This would solve https://github.com/coala/coala-bears/issues/929 .  I've created a GCI task to research this problem.

**Comment by Heredroky** [comment]
@jayvdb The goal is to convert the whole project to python 2?

**Comment by jayvdb** [comment]
> The goal is to convert the whole project to python 2?

No, the goal is to determine if `3to2` can be used to make coala run on Python 2, while continuing to use Python 3 syntax that isnt supported on Python 2.

I've been playing around with `guess-language` and `language-check`, and they both have the problem that `3to2` must be pre-installed, at least with the current version of `pip`.  https://github.com/myint/language-check/issues/36 .

**Comment by jack17529** [comment]
So you will need 2 things.
1. Use past in python then.
2. make code to integrate with guess-language and language-check.
